### PR TITLE
Feat/4506 jwt user mapping

### PR DESCRIPTION
**Pull request by morhag90**
auth-jwt user mapping

### CODE REVIEW

File: plugins/auth-jwt/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@amplication/plugin-auth-jwt",
-  "version": "1.4.0",
+  "version": "1.4.1-beta.4",
```
Comment (abrl91) [inline plugins/auth-jwt/package.json:3]: how come from 1.4.0 we jumped to 1.4.1-beta.4? I think it should be 1.4.1-beta.1

File: plugins/auth-jwt/package.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
   "author": "barshimi",
   "license": "ISC",
   "devDependencies": {
-    "@amplication/code-gen-types": "^2.0.1",
+    "@amplication/code-gen-types": "^2.0.2-beta.2",
```
Comment (abrl91) [inline plugins/auth-jwt/package.json:15]: same here

File: plugins/auth-jwt/src/core/create-auth-module.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+async function mapAuthModuleTemplate(
+  context: DsgContext,
+  templatePath: string,
+  fileName: string
+): Promise<Module> {
+  const { entities, resourceInfo, serverDirectories } = context;
+  const authEntity = entities?.find(
+    (x) => x.name === resourceInfo?.settings.authEntityName
+  );
+  if (!authEntity) throw OperationCanceledException; //todo: handle the exception
```
Comment (abrl91) [inline plugins/auth-jwt/src/core/create-auth-module.ts:36]: Where do we catch it?

File: plugins/auth-jwt/src/templates/jwt.strategy.template.ts [hunk 4]
```diff
@@ -1,14 +1,12 @@
...
+declare class ENTITY_SERVICE {}
+
 @Injectable()
 export class JwtStrategy extends JwtStrategyBase {
-  constructor(
-    protected readonly userService: UserService,
-    @Inject(JWT_SECRET_KEY) secretOrKey: string
-  ) {
-    super(userService, secretOrKey);
+  constructor(@Inject(JWT_SECRET_KEY) secretOrKey: string) {
```
Comment (abrl91) [inline plugins/auth-jwt/src/templates/jwt.strategy.template.ts:9]: Don't we need to inject the auth entity service?

File: plugins/auth-jwt/package.json (hunk 1 above)
Comment (morhag90) [reply plugins/auth-jwt/package.json:3]: @abrl91  After approve, I will update the version to 1.4.1

File: plugins/auth-jwt/package.json (hunk 2 above)
Comment (morhag90) [reply plugins/auth-jwt/package.json:15]: Same here = > version will be 2.0.2

File: plugins/auth-jwt/src/templates/jwt.strategy.template.ts (hunk 4 above)
Comment (morhag90) [reply plugins/auth-jwt/src/templates/jwt.strategy.template.ts:9]: @abrl91  Yes, it's implemented in create-jwt-strategy  

image

File: plugins/auth-jwt/src/core/create-auth-module.ts (hunk 3 above)
Comment (morhag90) [reply plugins/auth-jwt/src/core/create-auth-module.ts]: @abrl91 Update to error to log and stop the process

**Review by mulygottlieb — APPROVED** [review]

**Pull request merged**